src/MPDevice.cpp

```diff
@@ -4905,16 +4909,20 @@ bool MPDevice::readExportPayload(QJsonArray dataArray, QString &errorString)
 {
     /** Mooltiapp / Chrome App save file **/
 
+    const auto dataSize = dataArray.size();
+    const bool isBleExport = EXPORT_BLE_DATA_SIZE == dataArray.size() && dataArray[EXPORT_IS_BLE].toBool();
+    const QString deviceVersion = dataArray[EXPORT_DEVICE_VERSION].toString();
     /* Checks */
-    if (!((dataArray[9].toString() == "mooltipass" && dataArray.size() == 10) || (dataArray[9].toString() == "moolticute" && dataArray.size() == 14)))
+    if (!((deviceVersion == "mooltipass" && dataSize == EXPORT_MP_DATA_SIZE)
```

Comment:
EXPORT_MP_DATA_SIZE might be confusing here... Maybe rename is as "number of fields for moolticute export"?